src/en/charms-constraints.md

```diff
@@ -31,12 +31,24 @@ resulting system's resources will exceed the constraint-defined minimum.
 However, if the cloud cannot satisfy a constraint at all then an error will be
 emitted and a machine will not be provisioned.
 
-When using the localhost cloud, constraints are ineffectual due the nature of
-this cloud's underlying technology (LXD), where each machine will, by default,
-have access to **all** of the LXD host's resources. Here, an exact hardware
-specification can be requested, but is done at the LXD level (see example
-below).
+### Constraints and the LXD cloud
 
+For the 'lxd' cloud type (`v.2.4.1`) things are different. With this cloud,
+constraints are interpreted as resource maximums as opposed to minimums. In the
+absence of constraints, a machine (container) will, by default, have access to
+**all** of the underlying system's (LXD host) resources.
+
+This cloud type also honours instance constraints where 
+[AWS instance type][aws-instance-types-kirkland] names are used (e.g.
```

Comment:
Instance types from Azure and GCE are recognised as well.
The full list is here: https://github.com/dustinkirkland/instance-type/tree/master/yaml

Reply:
Does LXD pull this as a dep? Can we link to lxd vs a former employee's GH?